internal/versions/osimages.go

```diff
@@ -0,0 +1,202 @@
+package versions
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/go-openapi/swag"
+	"github.com/hashicorp/go-version"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+	"github.com/thoas/go-funk"
+)
+
+//go:generate mockgen --build_flags=--mod=mod -package versions -destination mock_osimages.go -self_package github.com/openshift/assisted-service/internal/versions . OSImages
+type OSImages interface {
```

Comment:
Nice:) It's a good a idea to extract these from Version. So maybe later we should rename 'Versions' to 'ReleaseImages' interface, to make it consistent.